india_compliance/gst_india/utils/gstr_1/gstr_1_data.py

```diff
@@ -171,16 +173,30 @@ def get_query_with_common_filters(self, query):
         return query
 
     def calculate_totals(self, query, si_doc, key):
-        # TODO: Handle Refunds and TDS
+        # TODO: Handle TDS
+        refund_amount = (
+            frappe.qb.from_(self.si_taxes)
+            .select(
+                Sum(self.si_taxes.base_tax_amount_after_discount_amount).as_(
+                    "refund_amount"
+                )
+            )
+            .where(self.si_taxes.gst_tax_type.isin(GST_REFUND_TAX_TYPES))
+            .where(self.si_taxes.parent == si_doc.name)
+        )
+
         return query.select(
-            IfNull(
-                Case()
-                .when(
-                    si_doc.base_rounded_total != 0,
-                    si_doc.base_rounded_total,
+            (
+                IfNull(
+                    Case()
+                    .when(
+                        si_doc.base_rounded_total != 0,
+                        si_doc.base_rounded_total,
+                    )
+                    .else_(si_doc.base_grand_total),
+                    0,
                 )
-                .else_(si_doc.base_grand_total),
-                0,
+                - IfNull(refund_amount, 0)
```

Comment:
Subquery as a Join is better than in select / where statements

https://stackoverflow.com/questions/2577174/join-vs-sub-query